### ISSUE TITLE
Make backdrop = 1.x optional

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#3571]
This issue can be summarised in the following:

Since we can do `dependencies[] = system (1.x)` or `dependencies[] = system (>=1.0.0, <2.0.0)`, is there any practical reason to keep `backdrop = 1.x` in .info files?

I have not fully fleshed this out yet, but here's a mind-dump for now:

Why should we be "forcing" contrib to create a 2.x branch, when it might be the case that some of these modules are so simple that even their 1.x versions could still keep working in Backdrop 2.0 (or Backdrop 3 for that matter)? In those cases, `dependencies[] = system (1.x)` is much more intuitive/flexible than `backdrop = 1.x`.

If we implement namespacing support for dependencies like D7.40 (#3570), then it could actually be `dependencies[] = backdrop:system (1.x)` (only for clarity and future-proofing; the D7 implementation of specifying project in `dependencies[]` is optional, so there is no BC breakage).

I was also thinking that at some point it would be awesome to support mutually exclusive dependencies (think `XOR` condition), so that contrib maintainers, or agencies that have custom modules that work with both Backdrop and D7 can maintain a single version, by specifying something like `dependencies[] = backdrop:system (1.x) ^ drupal:system (>=7.40)`.

Supporting mutually exclusive dependencies would also help in cases of contrib being merged into core. After #1376 gets implemented for example, contrib modules that were previously declaring a dependency on https://github.com/backdrop-contrib/field_formatter_settings would be able to declare something like `dependencies[] = field_formatter_settings ^ backdrop:system (>=1.13)`.
